sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaSource.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.tika;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.NoSuchElementException;
+
+import javax.annotation.Nullable;
+
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.io.BoundedSource;
+import org.apache.beam.sdk.io.FileBasedSource;
+import org.apache.beam.sdk.io.FileSystems;
+import org.apache.beam.sdk.io.Source;
+import org.apache.beam.sdk.io.fs.MatchResult;
+import org.apache.beam.sdk.io.fs.MatchResult.Metadata;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.joda.time.Instant;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation detail of {@link TikaIO.Read}.
+ *
+ * <p>A {@link Source} which can represent the content of the files parsed by Apache Tika.
+ */
+class TikaSource extends BoundedSource<String> {
```

Comment:
Same comment as for the `Reader`.

Reply:
Hi, I thought it could be reasonable but let me try to rearrange it a bit and see how it looks...

Reply:
I've done an initial inlining, out TikaReader inside TikaSource. As I said earlier, at the moment TikaSource has the internal FileBasedSource near duplicate of the core FilePatternReader. If you don't mind then I'd rather try to do another PR to get the core FilePatternReader refactored a bit such that it can work with TikaReader - then it will look a bit neater if I clean up TikaSource and then move it inside TikaIO

Reply:
Cool, thanks !